cmd/lncli/cmd_build_route.go

```diff
@@ -0,0 +1,116 @@
+// +build routerrpc
+
+package main
+
+import (
+	"context"
+	"errors"
+	"strconv"
+	"strings"
+
+	"github.com/lightningnetwork/lnd"
+	"github.com/lightningnetwork/lnd/lnrpc/routerrpc"
+	"github.com/lightningnetwork/lnd/routing/route"
+	"github.com/urfave/cli"
+)
+
+var buildRouteCommand = cli.Command{
+	Name:     "buildroute",
+	Category: "Payments",
+	Usage:    "Build a route from a list of channel ids and/or pubkeys.",
+	Action:   actionDecorator(buildRoute),
+	Flags: []cli.Flag{
+		cli.Int64Flag{
+			Name:  "amt",
+			Usage: "the amount to send expressed in satoshis",
+		},
+		cli.Int64Flag{
+			Name: "final_cltv_delta",
+			Usage: "number of blocks the last hop has to reveal " +
+				"the preimage",
+			Value: lnd.DefaultBitcoinTimeLockDelta,
+		},
+		cli.StringFlag{
+			Name:  "hops",
+			Usage: "comma separated channel ids and/or hex pubkeys",
```

Comment:
Shouldn't we just stick to pubkeys only? As chanids won't be guaranteed.

Reply:
Discussed offline and decided to stick to pubkeys only with an optional outgoing channel restriction for the first hop.